packages/python/plotly/plotly/express/_core.py

```diff
@@ -1599,6 +1603,31 @@ def aggfunc_continuous(x):
     return args
 
 
+def process_dataframe_timeline(args):
+    """
+    Massage input for bar traces for px.timeline()
+    """
+    args["is_timeline"] = True
+    if args["x_start"] is None or args["x_end"] is None:
+        raise ValueError("Both x_start and x_end are required")
+
+    try:
+        x_start = pd.to_datetime(args["data_frame"][args["x_start"]])
+        x_end = pd.to_datetime(args["data_frame"][args["x_end"]])
```

Comment:
There is some (for me ) unexpected behavior when the start and/or end times are plain numbers (parseable by pandas to_datetime). The default in pandas is to denominate in nano-seconds, whereas the subsequent time delta in process_dataframe_timeline is converted to mili-seconds.

If you try to plot something small such as [1, 2, 10], timeline will happily convert it and end up with all zeros and an empty plot since 1ns, 2ns and 10ns are all less than 1ms.

Should I create a bug report / enhancement for this issue?